src/hud.cpp

```diff
@@ -464,15 +503,71 @@ void Hud::drawCrosshair() {
 	}
 }
 
-
-void Hud::drawSelectionBoxes(std::vector<aabb3f> &hilightboxes) {
-	for (std::vector<aabb3f>::const_iterator
-			i = hilightboxes.begin();
-			i != hilightboxes.end(); ++i) {
-		driver->draw3DBox(*i, selectionbox_argb);
+void Hud::setHighlightedPos(const v3f &pos, const v3s16 &camera_offset)
+{
+	m_camera_offset = camera_offset;
+	m_highlighted_pos = pos;
+	m_highlighted_pos_with_offset = pos - intToFloat(camera_offset, BS);
+}
+	
+void Hud::drawSelectionMesh()
+{	
+	if (!m_use_selectionmesh) {
+		video::SMaterial oldmaterial = driver->getMaterial2D();
+		driver->setMaterial(m_selection_material);
+		for (std::vector<aabb3f>::const_iterator
+				i = m_selectionboxes.begin();
+				i != m_selectionboxes.end(); ++i) {
+			aabb3f box = aabb3f(
+				i->MinEdge + m_highlighted_pos_with_offset,
+				i->MaxEdge + m_highlighted_pos_with_offset);
+			driver->draw3DBox(box, selectionbox_argb);
+		}
+		driver->setMaterial(oldmaterial);
+	} else if (m_selectionmesh) {
```

Comment:
Just a question: When will m_use_selectionmesh be true but m_selectionmesh == NULL?